config/pmd/ruleset.xml

```diff
@@ -0,0 +1,79 @@
+<?xml version="1.0"?>
+
+<ruleset name="Java Logging ruleset"
+         xmlns="http://pmd.sourceforge.net/ruleset/2.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://pmd.sourceforge.net/ruleset/2.0.0 http://pmd.sourceforge.net/ruleset_2_0_0.xsd">
+    <description>Rules for Java Logging</description>
+
+    <rule ref="rulesets/java/android.xml" />
+    <rule ref="rulesets/java/basic.xml" />
+    <rule ref="rulesets/java/braces.xml" />
+    <rule ref="rulesets/java/clone.xml" />
+    <rule ref="rulesets/java/codesize.xml" />
+    <rule ref="rulesets/java/comments.xml">
+        <exclude name="CommentRequired" />
+        <exclude name="CommentSize" />
+    </rule>
+    <rule ref="rulesets/java/controversial.xml">
+        <exclude name="AtLeastOneConstructor" />
+        <exclude name="OnlyOneReturn" />
+    </rule>
+    <rule ref="rulesets/java/coupling.xml">
+        <exclude name="LoosePackageCoupling" />
+    </rule>
+    <rule ref="rulesets/java/coupling.xml/ExcessiveImports">
+        <properties>
+            <property name="minimum" value="35" /><!-- should be reduced -->
+        </properties>
+    </rule>
+    <rule ref="rulesets/java/coupling.xml/LawOfDemeter">
+        <properties>
+            <property name="violationSuppressRegex" value="(.*method chain calls.*|.*object not created locally.*)" />
+        </properties>
+    </rule>
+    <rule ref="rulesets/java/design.xml">
+        <exclude name="UncommentedEmptyMethodBody" />
+        <exclude name="UseUtilityClass" />
+    </rule>
+    <rule ref="rulesets/java/empty.xml" />
+    <rule ref="rulesets/java/finalizers.xml" />
+    <rule ref="rulesets/java/imports.xml">
+        <exclude name="TooManyStaticImports" />
+    </rule>
+    <rule ref="rulesets/java/j2ee.xml" />
+    <rule ref="rulesets/java/junit.xml">
+        <exclude name="JUnitTestContainsTooManyAsserts" />
+    </rule>
+    <rule ref="rulesets/java/logging-jakarta-commons.xml">
+        <exclude name="GuardLogStatement" />
+    </rule>
+    <rule ref="rulesets/java/logging-java.xml" />
+    <rule ref="rulesets/java/metrics.xml" />
+    <rule ref="rulesets/java/naming.xml">
+        <exclude name="LongVariable" />
```

Comment:
why?

Reply:
One of the variable is publicly accessible and is related to environment variable, the others are just normal vars e.g. `private final Supplier<String> requestIdGenerator;`

Setting can be adjusted to longer char sequence size and/or some variables can be changed. i chose an option to not change names already existing in codebase

Reply:
:+1:
17 is quite low now seeing a sample variable.